### PR TITLE
Contactlist2

### PR DESCRIPTION
**Pull request by SaltMining**

### CODE REVIEW

File: src/main/java/ConfirmationStage/ConfirmationStage.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    static {
+        try {
+            instance = new ConfirmationStage();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private ConfirmationStage () throws IOException {
```
Comment (rick2909) [inline src/main/java/ConfirmationStage/ConfirmationStage.java:21]: Kan deze verwijderd worden?

File: src/main/java/Main.java [hunk 2]
```diff
@@ -26,5 +26,4 @@ public static void main(String[] args) {
 // database.UserDatabase.createUser(String[], long);
 // database.UserDatabase.updateUser(int, String[], long);
 // new User("Tester","geheim123");
-// Company company = Company.createNewCompany(testCompanyInfo);
-
+// Company company = Company.createNewCompany(testCompanyInfo);
```
Comment (rick2909) [inline src/main/java/Main.java:29]: ALWAYS NEW LINE AT END OF FILE

File: src/main/java/SceneManager/ScenePile.java [hunk 3]
```diff
@@ -1,13 +1,14 @@
 package SceneManager;
 
 import javafx.stage.Stage;
+
```
Comment (rick2909) [inline src/main/java/SceneManager/ScenePile.java:4]: pls remove extra line

File: src/main/java/SceneManager/ScenePile.java [hunk 4]
```diff
@@ -28,12 +29,13 @@ public static void switchScene(String name) {
     }
```
Comment (rick2909) [inline src/main/java/SceneManager/ScenePile.java:31]: pls remove extra line

File: src/main/java/controllers/ContactCell.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.GridPane;
+import SceneManager.SceneMaker;
+import SceneManager.ScenePile;
+import model.Company;
+import model.User;
+
+import java.io.IOException;
+
+public class ContactCell extends ListCell<Object>  {
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:18]: 2 space before `{` remove one

File: src/main/java/controllers/ContactCell.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            if (loader == null) {
+                loader = new FXMLLoader(getClass().getResource("/ListCell.fxml"));
+                loader.setController(this);
+
+                try {
+                    loader.load();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:52]: remove extra line

File: src/main/java/controllers/ContactCell.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    private GridPane contactGrid;
+
+    private FXMLLoader loader;
+
+    @Override
+    protected void updateItem(Object contact, boolean empty) {
+        super.updateItem(contact, empty);
+
+        if(empty || contact == null) {
+
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:38]: remove line

File: src/main/java/controllers/ContactCell.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+            }
+
+//            if(contact.getIsCompany() == 1) {
+//                Image image = new Image(String.valueOf(getClass().getResource("/Icons/iconfinder_commerical-building_103266 (1).png")));
+//                typeImage.setImage(image);
+//            }
+//            else {
+//                Image image = new Image(String.valueOf(getClass().getResource("/Icons/iconfinder_user_1608727 (1).png")));
+//                typeImage.setImage(image);
+//            }
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:72]: if not used annymore it can be removed

File: src/main/java/controllers/ContactCell.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+                    new SceneMaker("contactpage");
+                    ScenePile.switchScene("contactpage");
+                } else{
+                    DetailCompanyController.setCompany((Company)contact);
+                    new SceneMaker("editCompany");
+                    ScenePile.switchScene("editCompany");
+                }
+            });
+        }
+
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:89]: remove xtra line

File: src/main/java/controllers/ContactCell.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
```
Comment (rick2909) [inline src/main/java/controllers/ContactCell.java:91]: ALWAYS EMPTY LINE AT END OF FILE

File: src/main/java/controllers/DetailContactController.java [hunk 11]
```diff
@@ -1,54 +1,56 @@
...
-        info[4] = Country.getText();
-        info[5] = Phone.getText();
-        return info;
-    }
 
-    public Long dateToLong(LocalDate date){
-        ZoneId zoneId = ZoneId.systemDefault();
-        return date.atStartOfDay(zoneId).toEpochSecond();
+
+
```
Comment (rick2909) [inline src/main/java/controllers/DetailContactController.java:42]: remove one line

File: src/main/java/controllers/LoginController.java [hunk 12]
```diff
@@ -57,12 +59,12 @@ public static User getUser()
...
+        //SceneMaker scene6 = new SceneMaker("addContact");
+        //SceneMaker scenesearch = new SceneMaker("searchPage");
+        SceneMaker scene2 = new SceneMaker("addcompany");
         //LoginScreen is initialized in main.
         SceneMaker scene4 = new SceneMaker("menu");
         SceneMaker scene5 = new SceneMaker("options");
     }
-}
-
+}
```
Comment (rick2909) [inline src/main/java/controllers/LoginController.java:70]: ALWAYS EXTRA LINE AD END OF FILE

File: src/main/java/model/Contact.java [hunk 13]
```diff
@@ -30,7 +30,7 @@ public void updateContact(Integer contactId, boolean isCompany, String[] info) {
         contactDatabase.updateContact(contactId,isCompany,info);
     }
 
-    public void deleteContact(Integer contactId) {
-        contactDatabase.deleteContact(contactId);
+    public void deleteContact(Integer userId, Integer contactId, boolean is_company) {
+        contactDatabase.deleteContact(userId, contactId, is_company);
     }
-}
+}
```
Comment (rick2909) [inline src/main/java/model/Contact.java:38]: ALWAYS EXTRA LINE AD END OF FILE

File: src/main/java/Main.java (hunk 2 above)
Comment (rick2909) [reply src/main/java/Main.java:29]: kan deze code vewijderd worden?

File: src/main/java/controllers/AddCompanyController.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.net.URL;
+import java.util.ResourceBundle;
+
+public class AddCompanyController implements Initializable {
+    @FXML
+    private JFXTextField Name, Email, Address, City, Country, Phone;
+    @FXML
+    private JFXButton addButton, backButton;
+    @FXML
```
Comment (joepschrama) [inline src/main/java/controllers/AddCompanyController.java:21]: whitelines between variables

File: src/main/java/controllers/AddContactController.java [hunk 15]
```diff
@@ -3,25 +3,40 @@
...
+    @FXML
+    private JFXTextField Name, Email, Address, City, Country, Phone;
+    @FXML
+    private JFXToggleButton Company;
+    @FXML
+    private JFXDatePicker BirthDate;
+    @FXML
+    private JFXButton Add, backButton;
+    @FXML
+    private Label ErrorLabel;
```
Comment (joepschrama) [inline src/main/java/controllers/AddContactController.java:25]: whitelines between variables

File: src/main/java/controllers/DetailContactController.java [hunk 16]
```diff
@@ -1,54 +1,56 @@
...
-import java.time.LocalDate;
-import java.time.ZoneId;
 import java.util.ResourceBundle;
 
 public class DetailContactController implements Initializable {
     @FXML private JFXTextField Name, Email, Address, City, Country, Phone;
-    @FXML private JFXCheckBox Company;
     @FXML private JFXDatePicker BirthDate;
-    @FXML private JFXButton Delete, Change;
+    @FXML private JFXButton Delete, backButton;
```
Comment (joepschrama) [inline src/main/java/controllers/DetailContactController.java:18]: @FXML above variable, not same line

File: src/main/java/controllers/LoginController.java [hunk 17]
```diff
@@ -43,6 +43,8 @@ private void handleButtonAction (ActionEvent event) throws IOException
         if(user != null) {
             URL url = new File("src/main/resources/HomeScreen.fxml").toURI().toURL();
             SceneMaker scene6 = new SceneMaker("usersettings");
+            SceneMaker scene1 = new SceneMaker("contactlist");
+            //initialize();
```
Comment (joepschrama) [inline src/main/java/controllers/LoginController.java:47]: comment niet nodig

File: src/main/java/controllers/LoginController.java [hunk 18]
```diff
@@ -57,12 +59,12 @@ public static User getUser()
 
     public void initialize() {
         SceneMaker scene = new SceneMaker("confirmation");
-        SceneMaker scene1 = new SceneMaker("contactlist");
-        SceneMaker scene2 = new SceneMaker("contactpage");
         SceneMaker scene3 = new SceneMaker("HomeScreen");
+        //SceneMaker scene6 = new SceneMaker("addContact");
+        //SceneMaker scenesearch = new SceneMaker("searchPage");
```
Comment (joepschrama) [inline src/main/java/controllers/LoginController.java:64]: comments niet nodig = weghalen

File: src/main/java/database/ContactDatabase.java [hunk 19]
```diff
@@ -138,11 +138,53 @@ public Integer addContact(Integer userId, Integer contactId, boolean is_company)
             System.out.println(e);
         }
 
+        //HashMap<String, Object> user
```
Comment (joepschrama) [inline src/main/java/database/ContactDatabase.java:141]: Comment niet nodig = weghalen

File: src/main/java/Main.java [hunk 20]
```diff
@@ -26,4 +26,4 @@ public static void main(String[] args) {
 // database.UserDatabase.createUser(String[], long);
 // database.UserDatabase.updateUser(int, String[], long);
 // new User("Tester","geheim123");
-// Company company = Company.createNewCompany(testCompanyInfo);
\ No newline at end of file
+// Company company = Company.createNewCompany(testCompanyInfo);
```
Comment (rick2909) [inline src/main/java/Main.java:29]: why is not all commended code deleted

Comment (Exerchan) [reply src/main/java/Main.java]: ur outdated. Get with the times old man

**Pull request merged**